embrace-android-core/src/main/kotlin/io/embrace/android/embracesdk/internal/logs/EmbraceLogService.kt

```diff
@@ -286,25 +270,23 @@ public class EmbraceLogService(
     }
 
     private fun trimToMaxLength(message: String): String {
-        val maxLength =
-            if (configService.appFramework == AppFramework.UNITY) {
-                LOG_MESSAGE_UNITY_MAXIMUM_ALLOWED_LENGTH
-            } else {
-                configService.logMessageBehavior.getLogMessageMaximumAllowedLength()
-            }
+        val maxLength = if (configService.appFramework == AppFramework.UNITY) {
+            LOG_MESSAGE_UNITY_MAXIMUM_ALLOWED_LENGTH
+        } else {
+            configService.logMessageBehavior.getLogMessageMaximumAllowedLength()
+        }
 
-        return if (message.length > maxLength) {
+        if (message.length > maxLength) {
+            logger.logWarning("Truncating message of ${message.length} characters to $maxLength characters")
             val endChars = "..."
 
-            // ensure that we never end up with a negative offset when extracting substring, regardless of the config value set
-            val allowedLength = when {
-                maxLength >= endChars.length -> maxLength - endChars.length
-                else -> LOG_MESSAGE_MAXIMUM_ALLOWED_LENGTH - endChars.length
+            if (maxLength <= endChars.length) {
```

Comment:
This takes care of the edge case, but effectively erases the value being set. Why not just return first the X characters you can based on max length and forget about the ...?

Reply:
I mainly wanted to leave the ... because it indicates that the value is being truncated. Your suggestion also makes sense though

Reply:
fixed!